samples/drivers/i2c_scanner/src/main.c

```diff
@@ -10,7 +10,11 @@
 #include <device.h>
 #include <i2c.h>
 
+#ifdef ARDUINO_I2C_LABEL
```

Comment:
This allows to at least provide a valid I2C reference for those who provide a arduino I2C.
This is better than defaulting to I2C_0. Hence I +1 the change.

Then, thinking a bit more ahead:
arduino I2C is made to be available connectors to add shields and then for boards with embedded sensors on I2C bus, this is not the bus used. So, for those boards, while this is a valid bus reference, it might be a I2C bus with no slaves, while there is another I2C bus more interesting to pay with. It would be an interesting change to scan all the available buses, but for now I don't see a simple way to detect that automatically since devices data miss a meta information to carry the type of the device. Maybe, in a latter iteration, the sample could be made interactive so the user provides the reference of the bus to scan?

Reply:
Making it interactive would be complex; ideally it would just scan and report on all available buses if no specific bus was identified at build time.